hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java

```diff
@@ -351,7 +351,7 @@ public ResponseCodeEnum adjustBalance(AccountID aId, TokenID tId, long adjustmen
 	public ResponseCodeEnum changeOwner(NftId nftId, AccountID from, AccountID to) {
 		final var tId = nftId.tokenId();
 		return sanityChecked(from, to, tId, token -> {
-			if (!nftsLedger.exists(nftId)) {
+			if (!nftId.isInternal() && !nftsLedger.exists(nftId)) {
```

Comment:
In the case for normal nft transfers, the NFT id will have `from`, `to` and `serialNumber` all valid. We should not include this check `!nftId.isInternal()` since `serialNumber` will not be `-1`

Reply:
I think we actually need this check `!nftId.isInternal()` here since that code base is also used for TokenUpdate operation where `serialNumber` is -1